tests/RomTest.php

```diff
@@ -710,68 +710,92 @@ public function testSetHeartColorsBlue()
     {
         $this->rom->setHeartColors('blue');
 
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA1E));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA20));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA22));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA24));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA26));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA28));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA2A));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA2C));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA2E));
-        $this->assertEquals(0x2C, $this->rom->read(0x6FA30));
-        $this->assertEquals(0x0D, $this->rom->read(0x65561));
+        $this->assertHeartColorSetting('blue');
```

Comment:
The reason I didn't make a helper function for the test was to keep everything contained within each unit test, while each setting does indeed change all the same values there's no reason that it wouldn't modify some other values in the future (for other aesthetic purposes), then we would have to  special case this assert function for that, also one is forced to scroll the file to see what is actually being asserted, makes the test a little harder to see what is being tested in the single context.

Reply:
I'm of two minds on this. I did the extraction since it made writing the additional tests easier, and in the case of the `red` and `default` tests, prevented a 100% copy-paste job.

You make valid points about encapsulation within the test itself, but I'd like to keep this as-is, with the caveat of "if we do end up having to special case this, then re-inline them all." Granted this test file is larger than what I'm used to (1k lines!), which is likely why I'm more predisposed to extracting common asserts/etc. out like this.

If you insist though I'll re-inline them all now. Let me know which way to go!